src/main/java/graphql/nadel/Nadel.java

```diff
@@ -97,47 +88,67 @@ public TypeDefinitionRegistry buildRegistry(ServiceDefinition serviceDefinition)
-//            ServiceDefinition serviceDefinition = this.stitchingDsl.getServiceByField().get(fieldDefinition);
-//            SchemaNamespace schemaNamespace = assertNotNull(this.stitchingDsl.getNamespaceByService().get(serviceDefinition.getName()));
-//            result.putIfAbsent(schemaNamespace, new ArrayList<>());
-//
-//
-//            String parentType = fieldTransformation.getParentDefinition().getName();
-//
-//            String originalFieldName = fieldDefinition.getName();
-//            Type targetType = assertNotNull(fieldTransformation.getTargetType());
-//            String targetTypeName = TypeInfo.typeInfo(targetType).getTypeName().getName();
-//            String newFieldName = fieldTransformation.getTargetName();
-//
-//            String queryField = targetTypeName.toLowerCase();
-//            SchemaNamespace targetNamespace = findNameSpace(targetTypeName);
-//            Link link = Link
-//                    .from(schemaNamespace, parentType, newFieldName, originalFieldName)
-//                    .to(targetNamespace, targetTypeName, queryField, "id")
-//                    .replaceFromField()
-//                    .build();
-//            result.get(schemaNamespace).add(link);
-//        });
-        return result;
+
+    private List<Link> createLinks(ServiceDefinition serviceDefinition) {
+        List<Link> links = new ArrayList<>();
+
+        SchemaNamespace schemaNamespace = assertNotNull(this.namespaceByService.get(serviceDefinition.getName()));
+
+        final List<FieldDefinitionWithParentType> defs = collectFieldTransformations(serviceDefinition);
+
+        for (FieldDefinitionWithParentType definition : defs) {
+            final FieldTransformation transformation = definition.field().getFieldTransformation();
+            if (transformation.getInnerServiceHydration() != null) {
+                final InnerServiceHydration hydration = transformation.getInnerServiceHydration();
+                final Link link = createHydrationLink(schemaNamespace, definition, hydration);
+                links.add(link);
+            } else if (transformation.getFieldMappingDefinition() != null) {
+                throw new InvalidDslException("Field mapping not implemented yet.", transformation.getSourceLocation());
+            }
+        }
+        return links;
+    }
+
+    private Link createHydrationLink(SchemaNamespace schemaNamespace, FieldDefinitionWithParentType definition,
+                                     InnerServiceHydration hydration) {
+        SchemaNamespace targetService = assertNotNull(this.namespaceByService.get(hydration.getServiceName()));
+        final FieldDefinition targetField = findTargetFieldForHydration(hydration);
+        //TODO: will not work for lists or non nullable types. does braid support this at all?
+        String targetTypeName = TypeInfo.typeInfo(targetField.getType()).getName();
```

Comment:
@andimarek do we need to change braid in order to support lists? or did i miss something?